### PR TITLE
Added yaml modernizer to remove deprecated elements from openedx.yaml

### PR DESCRIPTION
**Pull request by mraarif**
Relevant JIRA issue [here](https://openedx.atlassian.net/browse/BOM-1974).

### CODE REVIEW

File: tests/test_modernize_openedx_yaml.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class TestYamlModernizer(TestCase):
+
+    def setUp(self):
+        self.yaml_file = TestYamlModernizer._setup_local_copy("sample_openedx.yaml")
+
+    @staticmethod
+    def _setup_local_copy(file_name):
+        current_directory = os.path.dirname(__file__)
+        temp_file = os.path.join(current_directory, str(uuid.uuid4()) + ".yaml")
```
Comment (jmbowman) [inline tests/test_modernize_openedx_yaml.py:18]: This works fine, but in the future you may want to consider the [tmpdir](https://docs.pytest.org/en/stable/tmpdir.html) pytest fixture; probably a fair bit simpler to implement that way.  Don't need to manually invent a filename, automatically handles removing the file at the end of the test, etc.

File: edx_repo_tools/modernize_openedx_yaml.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import click
+import sys
+from ruamel.yaml import YAML
+
+DEPRECATED_FIELDS = ['owner', 'supporting_teams', 'track_pulls', 'track-pulls']
+
+
+class YamlModernizer:
```
Comment (jmbowman) [inline edx_repo_tools/modernize_openedx_yaml.py:8]: A docstring here or at the module level might be useful for future reference.  pylint would normally complain about this, but it looks like we haven't enabled it for this repo yet...that's a project for another time.

**Comment by nedbat** [comment]
Will reading and dumping the yaml file keep the comments in the file?

**Comment by nedbat** [comment]
I saw in https://github.com/edx/edx-enterprise/pull/924 that the comments are preserved.  I'm not sure how that magic happens, but it's good :)  Sorry for the noise.

**Comment by mraarif** [comment]
that might be because I've used `ruamel.yaml` instead of `PyYAML`

**Comment by jmbowman** [comment]
Correct, we picked that package previously for automated `.travis.yml` updates specifically because it preserves comments and such.  I think it forces whitespace to become consistent regardless of what it was before, but that's arguably a good thing to be standardizing anyway.

**Pull request merged**